scripts/nbd-info.nse

```diff
@@ -0,0 +1,189 @@
+      if not status then
+	return
+      end
+    end
+
+    comm:attach(name)
+  end
+end
+
+local function parse_args()
+  local args = {}
+
+  local arg = stdnse.get_script_args(SCRIPT_NAME .. ".export-names")
+  if not arg then
+    -- An empty string for an export name indicates to the server that
+    -- we wish to attach to the default export.
+    arg = {}
+  elseif type(arg) ~= table then
+    arg = {arg}
+  end
+  args.export_name = arg
+
+  return args
+end
+
+action = function(host, port)
+  local args = parse_args()
+
+  local comm = nbd.Comm:new(host, port)
+
+  local status = comm:connect(args)
+  if not status then
+     return nil
+  end
+
+  -- If the service supports an unrecognized negotiation, or the
+  -- oldstyle negotiation, there's no more information to be had.
+  if comm.protocol.negotiation == "unrecognized" or comm.protocol.negotiation == "oldstyle" then
+    -- Nothing to do.
+    comm:close()
+
+  -- If the service supports the (non-fixed) newstyle negotiation,
+  -- which should be very rare, we can only send a single option. That
+  -- option is the name of the export to which we'd like to attach.
+  elseif comm.protocol.negotiation == "newstyle" then
+    newstyle_connection(comm, args)
+
+  -- If the service supports the fixed newstyle negotiation, then we
+  -- can perform option haggling to wring additional information from
+  -- it.
+  elseif comm.protocol.negotiation == "fixed newstyle" then
+    enumerate_options(comm)
+    newstyle_connection(comm, args)
+
+  -- Otherwise, we've got a mismatch between the library and this script.
+  else
+    assert(false, "NBD library supports more negotiation styles than this script.")
+  end
+
+  -- Master output table.
+  local output = stdnse.output_table()
+
+  -- Format protocol information.
+  local tbl = {}
+  if comm.protocol.negotiation == "oldstyle" and comm.exports["(default)"] then
+    if comm.exports["(default)"].hflags & nbd.NBD.handshake_flags.FIXED_NEWSTYLE then
+      table.insert(tbl, "The server software appears to support fixed newstyle negotiation, but not on this port.")
+    end
+  end
+  table.insert(tbl, ("Negotiation: %s"):format(comm.protocol.negotiation))
+  table.insert(tbl, ("SSL/TLS Wrapped: %s"):format(comm.protocol.ssl_tls))
+
+  output["Protocol"] = tbl
+
+  -- Format exported block device information.
+  local tbl = {}
+  local no_shares = true
+  for name, info in pairs(comm.exports) do
+    local exp = {}
+    if type(info.size) == "number" then
+      table.insert(exp, ("Size: %d bytes"):format(info.size))
+    end
+
+    if type(info.tflags) == "table" then
+      local keys = {}
+      for k, _ in pairs(info.tflags) do
+	if k ~= "HAS_FLAGS" then
+	  table.insert(keys, k)
+	end
+      end
+      exp["Transmission Flags"] = keys
+    end
+
+    no_shares = false
+    tbl[name] = exp
+  end
+
+  if not no_shares then
+    output["Exported Block Devices"] = tbl
+  end
+
+  return output, stdnse.format_output(true, output)
```

Comment:
`stdnse.output_table` should not be filtered through `stdnse.format_output`. Just return it directly.